pandas/core/window.py

```diff
@@ -0,0 +1,1361 @@
+        elif com.is_integer_dtype(values.dtype):
+            values = values.astype(float)
+        elif com.is_timedelta64_dtype(values.dtype):
+            values = values.view('i8').astype(float)
+        else:
+            try:
+                values = values.astype(float)
+            except (ValueError, TypeError):
+                raise TypeError("cannot handle this type -> {0}".format(values.dtype))
+
+        if kill_inf:
+            values = values.copy()
+            values[np.isinf(values)] = np.NaN
+
+        return values
+
+    def _wrap_result(self, result, block=None, obj=None):
+        """ wrap a single result """
+
+        if obj is None:
+            obj = self._selected_obj
+        if isinstance(result, np.ndarray):
+
+            # coerce if necessary
+            if block is not None:
+                if com.is_timedelta64_dtype(block.values.dtype):
+                    result = pd.to_timedelta(result.ravel(),unit='ns').values.reshape(result.shape)
+
+            if result.ndim == 1:
+                from pandas import Series
+                return Series(result, obj.index, name=obj.name)
+
+            return type(obj)(result,
+                             index=obj.index,
+                             columns=block.columns)
+        return result
+
+    def _wrap_results(self, results, blocks, obj):
+        """
+        wrap the results
+
+        Paramters
+        ---------
+        results : list of ndarrays
+        blocks : list of blocks
+        obj : conformed data (may be resampled)
+        """
+
+        final = []
+        for result, block in zip(results, blocks):
+
+            result = self._wrap_result(result, block=block, obj=obj)
+            if result.ndim == 1:
+                return result
+            final.append(result)
+
+        if not len(final):
+            return obj.astype('float64')
+        return pd.concat(final,axis=1).reindex(columns=obj.columns)
+
+    def _center_window(self, result, window):
+        """ center the result in the window """
+        if self.axis > result.ndim-1:
+            raise ValueError("Requested axis is larger then no. of argument "
+                             "dimensions")
+
+        from pandas import Series, DataFrame
+        offset = _offset(window, True)
+        if offset > 0:
+            if isinstance(result, (Series, DataFrame)):
+                result = result.slice_shift(-offset, axis=self.axis)
+            else:
+                lead_indexer = [slice(None)] * result.ndim
+                lead_indexer[self.axis] = slice(offset, None)
+                result = np.copy(result[tuple(lead_indexer)])
+        return result
+
+    def aggregate(self, arg, *args, **kwargs):
+        result, how = self._aggregate(arg, *args, **kwargs)
+        if result is None:
+            return self.apply(arg, args=args, kwargs=kwargs)
+        return result
+
+    agg = aggregate
+
+class Window(_Window):
+    """
+    Provides rolling transformations.
```

Comment:
'rolling transformations' is not the most clarifying terminology I find, maybe adding `(aka moving windows)` ?